src/Microsoft.Framework.PackageManager/Algorithm/DepthFirstPostOrderWalker.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.PackageManager.Algorithm
+{
+    /// <summary>
+    /// Walk the graph in depth-first post order and generate result
+    /// </summary>
+    /// <typeparam name="TNode">Type of the graph node</typeparam>
+    /// <typeparam name="TResult">Type of the result</typeparam>
+    public class DepthFirstPostOrderWalker<TNode, TResult>
+    {
+        private readonly Func<TNode, IEnumerable<TResult>, TResult> _visit;
+        private readonly Func<TNode, IEnumerable<TNode>> _getChildren;
+
+        public DepthFirstPostOrderWalker(Func<TNode, IEnumerable<TResult>, TResult> visit,
```

Comment:
Is the `visit` param really a `visitor`? Not clear to me exactly what the pattern is.